docs/contributing/writing-instrumentation-module.md

```diff
@@ -360,3 +360,203 @@ For example:
+by refactoring them with the limitations described below.
+
+Once everything is "indy compatible", we can evaluate changing the default value of `otel.javaagent.experimental.indy`
+to `true` and make it non-experimental.
+
+### shared classes and common classloader
+
+By default, all the advices of an instrumentation module will be loaded into isolated classloaders,
+one per instrumentation module. Some instrumentations require to use a common classloader in order
+to preserve the semantics of `static` fields and to share classes.
+
+In order to load multiple `InstrumentationModule` implementations in the same classloader, you need to
+override the `ExperimentalInstrumentationModule#getModuleGroup` to return an identical value.
+
+### classes injected in application classloader
+
+Injecting classes in the application classloader is possible by implementing the
+`ExperimentalInstrumentationModule#injectedClassNames` method. All the class names listed by the
+returned value will be loaded in the application classloader instead of the agent or instrumentation
+module classloader.
+
+This allows for example to access package-private methods that would not be accessible otherwise.
+
+### advice method signatures
+
+With inlined advices, the advice method signature does not have any limitations. It enables referring
+to types in the OpenTelemetry API, SDK or instrumentation helper classes directly because those
+are injected in the target classloader or the bootstrap classloader for visibility.
+
+With indy advices however, the advice method signature can't introduce types that are only
+visible in the advice classes.
+In practice, two sets of types can be used in the advice method signature:
+- types from the JDK in the bootstrap classloader like `Object`
+- types already visible from the instrumented code
+
+For example, when trying to instrument a method that has `HttpServletRequest` as argument:
+- the `HttpServletRequest` type can be used as it's part of the instrumented code
+- the `HttpServletRequestScope` is part of instrumentation and has to be of type `Object` for both the enter return value and the exit argument.
+
+```java
+@Advice.OnMethodEnter(suppress = Throwable.class, inlined = false)
+public static Object onEnter(@Advice.Argument(1) HttpServletRequest request) {
+  // custom type from the instrumentation module, has to be returned as Object
+  return new HttpServletRequestScope(request);
+}
+
+@Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class, inlined = false)
+public static void onExit(@Advice.Argument(1) Object request,
+                          @Advice.Enter Object enterScope) {
+  // enter scope provided as Object so we have to check type and cast
+  if (!(enterScope instanceof HttpServletRequestScope)) {
+    return;
+  }
+  HttpServletRequestScope requestScope = (HttpServletRequestScope)enterScope;
+}
+```
+
+### advice local variables
```

Comment:
Keep the first letter capitalized? don't forget to modify same problem below.

Reply:
I don't think this is required IMHO as it refers here to the "general concept of advices" which is a common topic of instrumentation. Using `Advice` in the title could also be slightly confusing with the `@Advice` annotation which is provided by ByteBuddy to implement it.

However, on the other end with Spring AOP they also refer to `Advice` with a capital A to describe the way to instrument : https://docs.spring.io/spring-framework/reference/core/aop/ataspectj/advice.html

Reply:
capitalizing the first word in the headers would be consistent with other markdown in this repo

Reply:
can you update the other headings too? thanks